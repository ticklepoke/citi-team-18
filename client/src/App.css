@import "~antd/dist/antd.css";

.banner-gradient {
  background: rgb(12, 74, 137);
  background: linear-gradient(
    0deg,
    rgba(12, 74, 137, 1) 27%,
    rgba(46, 184, 240, 1) 100%
  );
  height: 50px;
  padding-left: 30px;
  padding-top: 2px;
}

.background-light-grey {
  background-color: #fff;
  min-height: calc(100vh - 50px);
}

.text-white {
  color: #fff;
}

.text-center {
  text-align: center;
}

.icon-large {
  height: 50px;
  font-size: 50px;
}

.justify-center {
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
}

.cursor-pointer {
  cursor: pointer;
}

.text-padding-top {
  padding-top: 20px;
}

.modal-button {
  background-color: #2392cc !important;
  height: 100%;
  color: white;
  border: none;
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
}

.modal-button-container {
  padding: 20px;
  width: 100%;
}

.modal-button:hover {
  background-color: #2392cc;
}
