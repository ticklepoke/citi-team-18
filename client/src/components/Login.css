.bg-main {
    background-image: url("../assets/bgmask.jpg");
    background-color: #000;
    background-size: cover;
    background-repeat: no-repeat;
    background-position-y: -30px;
    /* filter: brightness(0.4); */
    position: absolute;
    height: calc(100vh - 50px);
    width: 100%;
}

.card {
    width: 40%;
    margin: 20px;
    margin-top: 200px;
}

.form-login {
    margin: 20px;
}
.display-flex-card {
    display: flex;
    align-items: center;
    justify-content: flex-end;
}

.display-flex-end {
    display: flex;
    align-items: center;
    justify-content: center;
}

.cancel-button {
}

.code-input {
    margin-bottom: 15px;
    margin-top: 10px;
}

.sms-label {
    text-align: center;
}

.sms-form {
    margin-top: 30px;
}

.text-error {
    color: red;
    margin: 10px;
}
@media (max-width: 800px) {
    .card {
        width: 90%;
        margin-top: 20px;
    }

    .display-flex-card {
        justify-content: center;
    }

    .code-input {
        width: 100%;
        overflow: auto;
    }
}
