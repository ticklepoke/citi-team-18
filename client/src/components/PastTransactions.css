.display-flex-between {
    display: flex;
    justify-content: space-between;
    align-items: center;
}

.transaction-row {
    width: 100%;
}

.transaction-card {
    margin-top: 10px;
}
.label-amount {
    text-align: center;
}
.amount {
    text-align: center;
}

.text-center {
    text-align: center;
}
.col-center {
    display: flex;
    justify-content: center;
    align-items: center;
    flex-direction: column;
}

.text-bold {
    font-weight: bolder;
}
