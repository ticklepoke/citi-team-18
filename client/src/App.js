import React from "react";
import "./App.css";
import Router from "./Router";
const App = () => (
    <div className="App">
        <nav className="banner-gradient">
            <img
                src={require("./assets/citiwhite.png")}
                height="40px"
                alt="Citi"
            />
        </nav>

        <Router />
    </div>
);

export default App;
