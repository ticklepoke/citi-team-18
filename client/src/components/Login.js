import React, { Fragment, useState } from "react";
import "./Login.css";
import { Card, Steps, Form, Button, Input } from "antd";
import ReactCodeInput from "react-verification-code-input";
import axios from "axios";
import { Redirect } from "react-router-dom";

const { Step } = Steps;
const Login = () => {
    const [step, setStep] = useState(0);
    const [mobile, setMobile] = useState("+65");
    const [sms, setSMS] = useState("");
    const [loading, setloading] = useState(false);
    const [error, setError] = useState(false);

    const submitNumber = async () => {
        try {
            const res = await axios.post("blah", { data: mobile });

            if (res.success) {
                setStep(1);
            }
        } catch (err) {
            setError(true);
        }
    };

    const cancelStep = () => {
        setStep(0);
    };

    const handleMobileChange = (event) => {
        setMobile(event.target.value);
        console.log(event.target.value);
    };

    const submitSMS = async (event) => {
        setSMS(event);
        try {
            const res = await axios.post("sms", { data: sms });

            if (res.success) {
                return <Redirect to="/transaction" />;
            }
        } catch (err) {
            setError(true);
        }
    };

    const renderError = () => {
        if (error) {
            return <span className="text-error">Please try again</span>;
        } else {
            return null;
        }
    };

    const renderStep = () => {
        if (step === 0) {
            return (
                <Form className="form-login">
                    <Form.Item
                        label="Mobile Number"
                        name="mobileNumber"
                        rules={[
                            {
                                required: true,
                                message: "Please input your username!",
                            },
                        ]}
                    >
                        <Input
                            placeholder="Enter mobile"
                            value={mobile}
                            onChange={handleMobileChange}
                        />
                    </Form.Item>

                    <Form.Item>
                        <Button type="primary" onClick={submitNumber}>
                            Submit
                        </Button>
                    </Form.Item>
                </Form>
            );
        } else if (step === 1) {
            return (
                <div className="display-flex-center">
                    <div className="sms-form">
                        <label className="sms-label">
                            Please enter the 6-digit pin sent to your number
                        </label>
                        <ReactCodeInput
                            className="code-input"
                            onComplete={submitSMS}
                            autoFocus={true}
                            loading={loading}
                            fieldWidth={50}
                        />
                        <Button
                            className="cancel-button"
                            type="secondary"
                            onClick={cancelStep}
                        >
                            Cancel
                        </Button>
                    </div>
                </div>
            );
        }
    };
    return (
        <Fragment>
            <div className="bg-main"></div>
            <div className="display-flex-card">
                <Card className="card">
                    <Steps current={step}>
                        <Step key={0} title="Login"></Step>
                        <Step key={1} title="Verification"></Step>
                    </Steps>
                    {renderStep()}
                    {renderError()}
                </Card>
            </div>
        </Fragment>
    );
};

export default Login;
