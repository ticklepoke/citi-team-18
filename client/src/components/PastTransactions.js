import React, { useState, useEffect, Fragment } from "react";
import { Row, Col, Card } from "antd";
import { UserOutlined } from "@ant-design/icons";
import Avatar from "antd/lib/avatar/avatar";

import "./PastTransactions.css";
const pastTransactions = [
    {
        from: "You",
        to: "Lester",
        amount: 100,
        date: "12 Sep 2020",
    },
    {
        from: "Lester",
        to: "You",
        amount: 110,
        date: "2 Sep 2020",
    },
    {
        from: "You",
        to: "Nigel",
        amount: 1,
        date: "1 Sep 2020",
    },
    {
        from: "You",
        to: "Anthony",
        amount: 1,
        date: "1 Sep 2020",
    },
];
export default function PastTransactions(props) {
    const [transactions, setTransactions] = useState(pastTransactions);
    const renderAvatar = (You) => {
        if (You == "You") {
            return (
                <Avatar
                    size="large"
                    src="https://zos.alipayobjects.com/rmsportal/ODTLcjxAfvqbxHnVXCYX.png"
                />
            );
        } else {
            return <Avatar icon={<UserOutlined />} size="large" />;
        }
    };

    useEffect(() => {
        console.log(props);
    });

    const renderCards = () => {
        const cards = transactions.map((item) => {
            return (
                <Card className="transaction-card">
                    <Row
                        className="transaction-row"
                        align="top"
                        justify="center"
                    >
                        <Col span={4} className="col-center">
                            {renderAvatar(item.from)}
                            <span className="text-center">From: </span>
                            <h4 className="text-center text-bold">
                                {item.from}
                            </h4>
                        </Col>
                        <Col span={16}>
                            <h3 className="label-amount">Amount</h3>
                            <h2 className="amount">SGD {item.amount}</h2>
                            <h5 className="text-center">{item.date}</h5>
                        </Col>
                        <Col span={4} className="col-center">
                            {renderAvatar(item.to)}
                            <span className="text-center"> To: </span>
                            <h4 className="text-center text-bold">{item.to}</h4>
                        </Col>
                    </Row>
                </Card>
            );
        });

        return cards;
    };
    return (
        <div style={{ paddingTop: 20 }}>
            <strong>Past Transactions</strong>
            {renderCards()}
        </div>
    );
}
