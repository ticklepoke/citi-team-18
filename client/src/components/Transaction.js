import React from "react";
import { Button, Row, Col, Typography } from "antd";
import PastTransactions from "./PastTransactions";
import ButtonWithModal from "./ButtonWithModal";
import { HistoryOutlined, SettingOutlined } from "@ant-design/icons";
const { Title } = Typography;
const Transaction = () => (
    <Row style={{ paddingTop: 20 }} className="background-light-grey">
        <Col span={4}></Col>
        <Col span={16}>
            {" "}
            <Title level={2}> Welcome Back, Zhi Wei</Title>
            <Row
                style={{
                    width: "100%",
                    backgroundColor: "#2392cc",
                    minHeight: 200,
                    borderRadius: 14,
                    boxShadow:
                        "box-shadow: -4px 3px 24px 0px rgba(28,128,187,0.41);",
                }}
            >
                <Col
                    md={8}
                    sm={24}
                    className="justify-center modal-button-container"
                >
                    <ButtonWithModal />
                </Col>
                <Col
                    md={8}
                    sm={24}
                    className="justify-center cursor-pointer modal-button-container"
                >
                    <HistoryOutlined className="text-white icon-large" />
                    <h1 className="text-white text-center text-padding-top">
                        Past Transactions
                    </h1>
                </Col>
                <Col
                    md={8}
                    sm={24}
                    className="justify-center cursor-pointer modal-button-container"
                >
                    <SettingOutlined className="text-white icon-large" />
                    <h1 className="text-white text-center text-padding-top">
                        Settings
                    </h1>
                </Col>
            </Row>
        </Col>
        <Col span={4}></Col>
        <Col span={4}></Col>
        <Col span={16}>
            <PastTransactions />
        </Col>
        <Col span={4}></Col>
    </Row>
);

export default Transaction;
