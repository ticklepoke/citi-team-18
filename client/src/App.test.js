import React from "react";
import { render } from "@testing-library/react";
import App from "./App";
import Enzyme, { shallow } from "enzyme";

import Adapter from "enzyme-adapter-react-16";
import Login from "./components/Login";
import Transaction from "./components/Transaction";
import PastTransactions from "./components/PastTransactions";
Enzyme.configure({ adapter: new Adapter() });

test("renders app", () => {
    // const { getByText } = render(<App />);
    // const linkElement = getByText(/learn react/i);
    // expect(linkElement).toBeInTheDocument();

    const component = shallow(<App />);
    expect(component.getElements()).toMatchSnapshot();
});

test("renders login", () => {
    const component = shallow(<Login />);
    expect(component.getElements()).toMatchSnapshot();
});

test("renders transaction", () => {
    const component = shallow(<Transaction />);
    expect(component.getElements()).toMatchSnapshot();
});

test("renders past transactions", () => {
    const component = shallow(<PastTransactions />);
    expect(component.getElements()).toMatchSnapshot();
});
