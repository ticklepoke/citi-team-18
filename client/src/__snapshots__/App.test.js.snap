// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders app 1`] = `
Array [
  <div
    className="App"
  >
    <nav
      className="banner-gradient"
    >
      <img
        alt="Citi"
        height="40px"
        src="citiwhite.png"
      />
    </nav>
    <Router />
  </div>,
]
`;

exports[`renders login 1`] = `
Array [
  <React.Fragment>
    <div
      className="bg-main"
    />
    <div
      className="display-flex-card"
    >
      <Card
        className="card"
      >
        <Steps
          current={0}
        >
          <Step
            title="Login"
          />
          <Step
            title="Verification"
          />
        </Steps>
        <ForwardRef(InternalForm)
          className="form-login"
        >
          <FormItem
            label="Mobile Number"
            name="mobileNumber"
            rules={
              Array [
                Object {
                  "message": "Please input your username!",
                  "required": true,
                },
              ]
            }
          >
            <Input
              onChange={[Function]}
              placeholder="Enter mobile"
              type="text"
              value="+65"
            />
          </FormItem>
          <FormItem>
            <ForwardRef(InternalButton)
              block={false}
              disabled={true}
              ghost={false}
              htmlType="button"
              loading={false}
              onClick={[Function]}
              type="primary"
            >
              Submit
            </ForwardRef(InternalButton)>
          </FormItem>
        </ForwardRef(InternalForm)>
      </Card>
    </div>
  </React.Fragment>,
]
`;

exports[`renders past transactions 1`] = `
Array [
  <div
    style={
      Object {
        "paddingTop": 20,
      }
    }
  >
    <strong>
      Past Transactions
    </strong>
    <Card
      className="transaction-card"
    >
      <ForwardRef
        align="top"
        className="transaction-row"
        justify="center"
      >
        <ForwardRef
          className="col-center"
          span={4}
        >
          <ForwardRef(InternalAvatar)
            shape="circle"
            size="large"
            src="https://zos.alipayobjects.com/rmsportal/ODTLcjxAfvqbxHnVXCYX.png"
          />
          <span
            className="text-center"
          >
            From: 
          </span>
          <h4
            className="text-center text-bold"
          >
            You
          </h4>
        </ForwardRef>
        <ForwardRef
          span={16}
        >
          <h3
            className="label-amount"
          >
            Amount
          </h3>
          <h2
            className="amount"
          >
            SGD 
            100
          </h2>
          <h5
            className="text-center"
          >
            12 Sep 2020
          </h5>
        </ForwardRef>
        <ForwardRef
          className="col-center"
          span={4}
        >
          <ForwardRef(InternalAvatar)
            icon={<ForwardRef(UserOutlined) />}
            shape="circle"
            size="large"
          />
          <span
            className="text-center"
          >
             To: 
          </span>
          <h4
            className="text-center text-bold"
          >
            Lester
          </h4>
        </ForwardRef>
      </ForwardRef>
    </Card>
    <Card
      className="transaction-card"
    >
      <ForwardRef
        align="top"
        className="transaction-row"
        justify="center"
      >
        <ForwardRef
          className="col-center"
          span={4}
        >
          <ForwardRef(InternalAvatar)
            icon={<ForwardRef(UserOutlined) />}
            shape="circle"
            size="large"
          />
          <span
            className="text-center"
          >
            From: 
          </span>
          <h4
            className="text-center text-bold"
          >
            Lester
          </h4>
        </ForwardRef>
        <ForwardRef
          span={16}
        >
          <h3
            className="label-amount"
          >
            Amount
          </h3>
          <h2
            className="amount"
          >
            SGD 
            110
          </h2>
          <h5
            className="text-center"
          >
            2 Sep 2020
          </h5>
        </ForwardRef>
        <ForwardRef
          className="col-center"
          span={4}
        >
          <ForwardRef(InternalAvatar)
            shape="circle"
            size="large"
            src="https://zos.alipayobjects.com/rmsportal/ODTLcjxAfvqbxHnVXCYX.png"
          />
          <span
            className="text-center"
          >
             To: 
          </span>
          <h4
            className="text-center text-bold"
          >
            You
          </h4>
        </ForwardRef>
      </ForwardRef>
    </Card>
    <Card
      className="transaction-card"
    >
      <ForwardRef
        align="top"
        className="transaction-row"
        justify="center"
      >
        <ForwardRef
          className="col-center"
          span={4}
        >
          <ForwardRef(InternalAvatar)
            shape="circle"
            size="large"
            src="https://zos.alipayobjects.com/rmsportal/ODTLcjxAfvqbxHnVXCYX.png"
          />
          <span
            className="text-center"
          >
            From: 
          </span>
          <h4
            className="text-center text-bold"
          >
            You
          </h4>
        </ForwardRef>
        <ForwardRef
          span={16}
        >
          <h3
            className="label-amount"
          >
            Amount
          </h3>
          <h2
            className="amount"
          >
            SGD 
            1
          </h2>
          <h5
            className="text-center"
          >
            1 Sep 2020
          </h5>
        </ForwardRef>
        <ForwardRef
          className="col-center"
          span={4}
        >
          <ForwardRef(InternalAvatar)
            icon={<ForwardRef(UserOutlined) />}
            shape="circle"
            size="large"
          />
          <span
            className="text-center"
          >
             To: 
          </span>
          <h4
            className="text-center text-bold"
          >
            Nigel
          </h4>
        </ForwardRef>
      </ForwardRef>
    </Card>
    <Card
      className="transaction-card"
    >
      <ForwardRef
        align="top"
        className="transaction-row"
        justify="center"
      >
        <ForwardRef
          className="col-center"
          span={4}
        >
          <ForwardRef(InternalAvatar)
            shape="circle"
            size="large"
            src="https://zos.alipayobjects.com/rmsportal/ODTLcjxAfvqbxHnVXCYX.png"
          />
          <span
            className="text-center"
          >
            From: 
          </span>
          <h4
            className="text-center text-bold"
          >
            You
          </h4>
        </ForwardRef>
        <ForwardRef
          span={16}
        >
          <h3
            className="label-amount"
          >
            Amount
          </h3>
          <h2
            className="amount"
          >
            SGD 
            1
          </h2>
          <h5
            className="text-center"
          >
            1 Sep 2020
          </h5>
        </ForwardRef>
        <ForwardRef
          className="col-center"
          span={4}
        >
          <ForwardRef(InternalAvatar)
            icon={<ForwardRef(UserOutlined) />}
            shape="circle"
            size="large"
          />
          <span
            className="text-center"
          >
             To: 
          </span>
          <h4
            className="text-center text-bold"
          >
            Anthony
          </h4>
        </ForwardRef>
      </ForwardRef>
    </Card>
  </div>,
]
`;

exports[`renders transaction 1`] = `
Array [
  <ForwardRef
    className="background-light-grey"
    style={
      Object {
        "paddingTop": 20,
      }
    }
  >
    <ForwardRef
      span={4}
    />
    <ForwardRef
      span={16}
    >
       
      <div
        className="justify-center-around"
      >
        <Title
          level={2}
        >
           Welcome Back, Zhi Wei
        </Title>
        <ForwardRef(InternalButton)
          block={false}
          ghost={false}
          htmlType="button"
          loading={false}
          onClick={[Function]}
          style={
            Object {
              "borderRadius": 6,
              "padding": "0 15px 0 15px",
            }
          }
          type="primary"
        >
          Logout
        </ForwardRef(InternalButton)>
      </div>
      <ForwardRef
        style={
          Object {
            "backgroundColor": "#2392cc",
            "borderRadius": 14,
            "boxShadow": "-4px 3px 16px -4px rgba(35,146,204,0.86)",
            "marginTop": 30,
            "marginVertical": 40,
            "minHeight": 250,
            "width": "100%",
          }
        }
      >
        <ForwardRef
          style={
            Object {
              "flex": 1,
              "flexDirection": "column",
              "padding": 20,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#F6F7EB",
                "fontSize": 28,
                "marginTop": 10,
                "position": "relative",
                "top": 10,
              }
            }
          >
            Account Balance
          </Text>
          <Text
            level={2}
            style={
              Object {
                "color": "#F6F7EB",
                "fontSize": 67,
                "marginTop": 0,
              }
            }
          >
             
            $1890.30
             
          </Text>
        </ForwardRef>
      </ForwardRef>
      <NewTransactionButton />
    </ForwardRef>
    <ForwardRef
      span={4}
    />
    <ForwardRef
      span={4}
    />
    <ForwardRef
      span={16}
    >
      <PastTransactions />
    </ForwardRef>
    <ForwardRef
      span={4}
    />
  </ForwardRef>,
]
`;
